etl/generative_model_pipelines/open_ai_generative_model_pipeline.py

```diff
@@ -14,28 +14,35 @@
 )
 
 
-class OpenAiGenerativeModelResource(ConfigurableResource):  # type: ignore
-    """A ConfigurableResource to enrich Records using OpenAI's generative AI models."""
+class OpenAiGenerativeModelPipeline(GenerativeModelPipeline):
+    """
+    A concrete implementation of GenerativeModelPipeline.
 
-    openai_resource_params: OpenAiResourceParams
-    __template = """
+    Uses OpenAI's generative models to enrich Records.
+    """
+
+    def __init__(self, openai_pipeline_config: OpenAiPipelineConfig) -> None:
+        self.__openai_pipeline_config: OpenAiPipelineConfig = openai_pipeline_config
+        self.__template = """
                 Keep the answer as concise as possible.
```

Comment:
These multi-line strings mean you end up sending a bunch of whitespace and a leading newline. You can do 

```
self.__template = """\
Dedent all the way to the first column of the file.
"""
```

The `\` there means "don't include a leading newline".

Reply:
Thank you, I didn't know that. I have added the backslash.